src/code/z_lifemeter.c

```diff
@@ -1,17 +1,446 @@
+
+    interfaceCtx->heartsPrimR[0] = 0xFF;
+    interfaceCtx->heartsPrimG[0] = 0x46;
+    interfaceCtx->heartsPrimB[0] = 0x32;
+
+    interfaceCtx->heartsEnvR[0] = 0x32;
+    interfaceCtx->heartsEnvG[0] = 0x28;
+    interfaceCtx->heartsEnvB[0] = 0x3C;
+
+    interfaceCtx->heartsPrimR[1] = sHeartsPrimColors[type][0];
+    interfaceCtx->heartsPrimG[1] = sHeartsPrimColors[type][1];
+    interfaceCtx->heartsPrimB[1] = sHeartsPrimColors[type][2];
+
+    interfaceCtx->heartsEnvR[1] = sHeartsEnvColors[type][0];
+    interfaceCtx->heartsEnvG[1] = sHeartsEnvColors[type][1];
+    interfaceCtx->heartsEnvB[1] = sHeartsEnvColors[type][2];
+
+    rFactor = sHeartsPrimFactors[0][0] * factorBeating;
+    gFactor = sHeartsPrimFactors[0][1] * factorBeating;
+    bFactor = sHeartsPrimFactors[0][2] * factorBeating;
+
+    interfaceCtx->beatingHeartPrim[0] = (u8)(rFactor + 0xFF) & 0xFF;
+    interfaceCtx->beatingHeartPrim[1] = (u8)(gFactor + 0x46) & 0xFF;
+    interfaceCtx->beatingHeartPrim[2] = (u8)(bFactor + 0x32) & 0xFF;
+
+    rFactor = sHeartsEnvFactors[0][0] * factorBeating;
+    gFactor = sHeartsEnvFactors[0][1] * factorBeating;
+    bFactor = sHeartsEnvFactors[0][2] * factorBeating;
+
+    if (1) {}
+    ddType = type;
+
+    interfaceCtx->beatingHeartEnv[0] = (u8)(rFactor + 0x32) & 0xFF;
+    interfaceCtx->beatingHeartEnv[1] = (u8)(gFactor + 0x28) & 0xFF;
+    interfaceCtx->beatingHeartEnv[2] = (u8)(bFactor + 0x3C) & 0xFF;
+
+    sHeartsDDPrim[0][0] = 0xFF;
+    sHeartsDDPrim[0][1] = 0xFF;
+    sHeartsDDPrim[0][2] = 0xFF;
+
+    sHeartsDDEnv[0][0] = 0xC8;
+    sHeartsDDEnv[0][1] = 0x00;
+    sHeartsDDEnv[0][2] = 0x00;
+
+    sHeartsDDPrim[1][0] = sHeartsDDPrimColors[ddType][0];
+    sHeartsDDPrim[1][1] = sHeartsDDPrimColors[ddType][1];
+    sHeartsDDPrim[1][2] = sHeartsDDPrimColors[ddType][2];
+
+    sHeartsDDEnv[1][0] = sHeartsDDEnvColors[ddType][0];
+    sHeartsDDEnv[1][1] = sHeartsDDEnvColors[ddType][1];
+    sHeartsDDEnv[1][2] = sHeartsDDEnvColors[ddType][2];
+
+    rFactor = sHeartsDDPrimFactors[ddType][0] * ddFactor;
+    gFactor = sHeartsDDPrimFactors[ddType][1] * ddFactor;
+    bFactor = sHeartsDDPrimFactors[ddType][2] * ddFactor;
+
+    sBeatingHeartsDDPrim[0] = (u8)(rFactor + 0xFF) & 0xFF;
+    sBeatingHeartsDDPrim[1] = (u8)(gFactor + 0xFF) & 0xFF;
+    sBeatingHeartsDDPrim[2] = (u8)(bFactor + 0xFF) & 0xFF;
+
+    rFactor = sHeartsDDEnvFactors[ddType][0] * ddFactor;
+    gFactor = sHeartsDDEnvFactors[ddType][1] * ddFactor;
+    bFactor = sHeartsDDEnvFactors[ddType][2] * ddFactor;
+
+    sBeatingHeartsDDEnv[0] = (u8)(rFactor + 0xC8) & 0xFF;
+    sBeatingHeartsDDEnv[1] = (u8)(gFactor + 0x00) & 0xFF;
+    sBeatingHeartsDDEnv[2] = (u8)(bFactor + 0x00) & 0xFF;
+}
+
+s32 func_80100A80(GlobalContext* globalCtx) {
+    gSaveContext.health = (s16)globalCtx->interfaceCtx.unk_24E;
+    return 1;
+}
+
+s32 func_80100AA0(GlobalContext* globalCtx) {
+    InterfaceContext* interfaceCtx = &globalCtx->interfaceCtx;
+    interfaceCtx->unk_250 = 0x140;
```

Comment:
```suggestion
    InterfaceContext* interfaceCtx = &globalCtx->interfaceCtx;
    
    interfaceCtx->unk_250 = 0x140;
```

Reply:
The reason for this is our coding style separates the variable's declarations from the rest of the code.